src/actions/breakpoints/syncBreakpoint.js

```diff
@@ -106,98 +101,74 @@ export async function syncBreakpointPromise(
 
   const generatedSource = getSource(getState(), generatedSourceId);
 
-  if (!source) {
-    return null;
+  if (!source || !generatedSource) {
+    throw new Error(
+      `Could not find relevant sources ${sourceId} - ${generatedSourceId}`
+    );
   }
 
-  const { location, astLocation } = pendingBreakpoint;
+  const { location, generatedLocation, astLocation } = pendingBreakpoint;
   const previousLocation = { ...location, sourceId };
 
-  const scopedLocation = await makeScopedLocation(
+  const newLocation = await findNewLocation(
```

Comment:
Should we always be calling this here, or should we call `findBreakpointPosition` with `previousLocation` position, and fall back to the AST location as a second `findBreakpointPosition` call if nothing was found?

Reply:
I don't think this is strictly related to your changes, so it might be fine for now. My concern is that `findNewLocation` indiscriminantly does
```
  if (func) {
    line = func.location.start.line + offset.line;
  }
```
What if it changes the line to something that is actually wrong, and the existing `line` was actually what we should have checked?

Reply:
can you give me an example?

Reply:
Alright say I have
```
function foo() {
  var value = "bar";
  console.log(value);
}
```
with a breakpoint at the start of `console`. It's a breakpoint on line 2, or maybe 3 of the function at column 2.

Say I change the file to
```
function doThing() {
  var value = "bar";
  console.log("bar");
}
function foo() {
  doThing();
}
```
the current logic, I _think_, will take the generated location of the second line of `foo`, which is now the closing `}`, and fail to match, while leaving the breakpoint at the exact original location would have been fine.

It's all edgecasey, so honestly don't worry about it, especially since it's a existing behavior.